### PR TITLE
[#10581] Add top level domain hint for invalid email error

### LINKED ISSUE #10581: Update error message for incorrect email format
- **Environment**: V7.0.2

image
Error message for incorrect email should be updated to include check for top level domain

### PR DESCRIPTION
**Pull request by moziliar**
Fixes #10581

### CODE REVIEW

**Review by madanalogy — CHANGES_REQUESTED** [review]
Seems like your code isn't passing the lint or component tests on the CI. You need to resolve those issues. Do ask for help if you need it. 

Also, once you're done making changes, just remember that you'll need to re-request for a review. Otherwise, your PR will not be attended to. Feel free to tag me with a mention if you have problems requesting for a review.

**Comment by moziliar** [comment]
> Seems like your code isn't passing the lint or component tests on the CI. You need to resolve those issues. Do ask for help if you need it.
> 
> Also, once you're done making changes, just remember that you'll need to re-request for a review. Otherwise, your PR will not be attended to. Feel free to tag me with a mention if you have problems requesting for a review.

@madanalogy Sorry for leaving this PR for quite a while. I was told that everyone is supposed to do only 1 trivial (first-timer) issue, hence I held off from updating this PR. But I will just finish this off first then.

**Review by madanalogy — APPROVED** [review]
LGTM

**Review by t-cheepeng — APPROVED** [review]
LGTM

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 1]
```diff
@@ -146,7 +146,8 @@
 
     // field-specific error messages
     public static final String HINT_FOR_CORRECT_EMAIL =
-            "An email address contains some text followed by one '@' sign followed by some more text. "
+            "An email address contains some text followed by one '@' sign followed by some more text, and should end "
+                    + "with a top domain address like .com. "
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/util/FieldValidator.java:150]: "top domain" has no meaning. Top-level domain (TLD) is the universally understood term for what is mentioned here.

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]

**Comment by madanalogy** [comment]
@moziliar there seems to have been some changes to the test files, please help to resolve the conflicts thanks

**Review by madanalogy — APPROVED** [review]

**Pull request merged**